### PR TITLE
Add desktop implementations

### LINKED ISSUE #87: Windows, macOS, and Linux Support
**Is your feature request related to a problem? Please describe.**
Right now, this is the only plugin that is blocking us from release for desktop. Please consider adding support for windows, linux and macOS.

**Describe the solution you'd like**
macOS and Linux should just return false as they don't have real touch support natively, and windows should handle the soft keyboard.

### PR DESCRIPTION
**Pull request by cbenhagen**
For now all implementations return false.

See #87

### CODE REVIEW

**Comment by MisterJimson** [comment]
Dude this is awesome! Will take a look asap.

**Comment by MisterJimson** [comment]
Trying to test macos desktop.

I've changed `flutter_keyboard_visibility/pubspec.yaml` to point to path deps so I can `pub get`.

`flutter run -d macos` results in:

```
➜  example git:(desktop) ✗ flutter run -d macos
Launching lib/main.dart on macOS in debug mode...
Running pod install...                                             722ms
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:arm64, id:00006001-000459DE3441801E }
{ platform:macOS, arch:x86_64, id:00006001-000459DE3441801E }
.dart_tool/flutter_build/dart_plugin_registrant.dart:22:9: Error: 'FlutterKeyboardVisibilityPlugin' is imported from both 'package:flutter_keyboard_visibility_linux/flutter_keyboard_visibility_linux.dart' and 'package:flutter_keyboard_visibility_macos/flutter_keyboard_visibility_macos.dart'.
        FlutterKeyboardVisibilityPlugin.registerWith();
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.dart_tool/flutter_build/dart_plugin_registrant.dart:33:9: Error: 'FlutterKeyboardVisibilityPlugin' is imported from both 'package:flutter_keyboard_visibility_linux/flutter_keyboard_visibility_linux.dart' and 'package:flutter_keyboard_visibility_macos/flutter_keyboard_visibility_macos.dart'.
        FlutterKeyboardVisibilityPlugin.registerWith();
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.dart_tool/flutter_build/dart_plugin_registrant.dart:44:9: Error: 'FlutterKeyboardVisibilityPlugin' is imported from both 'package:flutter_keyboard_visibility_linux/flutter_keyboard_visibility_linux.dart' and 'package:flutter_keyboard_visibility_macos/flutter_keyboard_visibility_macos.dart'.
        FlutterKeyboardVisibilityPlugin.registerWith();
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Command PhaseScriptExecution failed with a nonzero exit code
warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Flutter Assemble' from project 'Runner')
warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'Runner' from project 'Runner')
** BUILD FAILED **

Building macOS application...                                           
Exception: Build process failed
```

**Comment by MisterJimson** [comment]
From this doc page: https://docs.flutter.dev/development/packages-and-plugins/developing-packages#dart-only-platform-implementations

It seems like each plugin should use a different class name. So `FlutterKeyboardVisibilityPlugin` -> `FlutterKeyboardVisibilityPluginMacos` for the macos package

**Comment by cbenhagen** [comment]
Strange, I wonder why this worked on my machine. But renaming is for sure a good idea. I also updated the README.

File: flutter_keyboard_visibility_linux/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Flutter Keyboard Visibility Linux
+Web support for Flutter Keyboard Visibility
```
Comment (MisterJimson) [inline flutter_keyboard_visibility_linux/README.md:2]:
```suggestion
Linux support for Flutter Keyboard Visibility
```

File: flutter_keyboard_visibility_linux/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  plugin:
+    implements: flutter_keyboard_visibility
+    platforms:
+      linux:
+        dartPluginClass: FlutterKeyboardVisibilityPluginLinux
+
+dependencies:
+  flutter_keyboard_visibility_platform_interface: ^2.0.0
+  flutter_web_plugins:
+    sdk: flutter
```
Comment (MisterJimson) [inline flutter_keyboard_visibility_linux/pubspec.yaml:21]: Is this needed?

File: flutter_keyboard_visibility_macos/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Flutter Keyboard Visibility Web
+Web support for Flutter Keyboard Visibility
```
Comment (MisterJimson) [inline flutter_keyboard_visibility_macos/README.md:2]:
```suggestion
macOS support for Flutter Keyboard Visibility
```

File: flutter_keyboard_visibility_windows/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Flutter Keyboard Visibility Web
```
Comment (MisterJimson) [inline flutter_keyboard_visibility_windows/README.md:1]:
```suggestion
# Flutter Keyboard Visibility Windows
```

File: flutter_keyboard_visibility_windows/README.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+Web support for Flutter Keyboard Visibility
```
Comment (MisterJimson) [inline flutter_keyboard_visibility_windows/README.md:2]:
```suggestion
Windows support for Flutter Keyboard Visibility
```

File: flutter_keyboard_visibility_windows/pubspec.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  plugin:
+    implements: flutter_keyboard_visibility
+    platforms:
+      windows:
+        dartPluginClass: FlutterKeyboardVisibilityPluginWindows
+
+dependencies:
+  flutter_keyboard_visibility_platform_interface: ^2.0.0
+  flutter_web_plugins:
+    sdk: flutter
```
Comment (MisterJimson) [inline flutter_keyboard_visibility_windows/pubspec.yaml:21]: Is this needed?

**Review by MisterJimson — CHANGES_REQUESTED** [review]
Awesome work, really appreciate it. Some minor changes and questions.

**Review by MisterJimson — APPROVED** [review]

**Comment by cbenhagen** [comment]
@MisterJimson sorry I had to pause my work on this for a moment. Thanks for finishing the PR!

**Comment by MisterJimson** [comment]
No worries at all!

**Pull request merged**